client/state/sites/hooks/use-premium-global-styles.ts

```diff
@@ -0,0 +1,78 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type GlobalStylesInfo = {
+	shouldLimitGlobalStyles: boolean;
+	globalStylesInUse: boolean;
+};
+
+type Site = {
+	isCreated: boolean;
+	id: number;
+};
+
+const NEW_SITE_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	// The next line should be replaced with true once the Gating global styles feature is live.
+	// That will make all non-created sites to limit global styles.
+	shouldLimitGlobalStyles: isEnabled( 'limit-global-styles' ),
+	globalStylesInUse: false,
+};
+
+// While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
+// showing notices for paid sites until we fetch the data from the server.
+const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	shouldLimitGlobalStyles: false,
+	globalStylesInUse: false,
+};
+
+const getGSInfoEndpointForSite = ( siteId: number ) => ( {
+	path: `sites/${ siteId }/global-styles-info`,
+	apiNamespace: 'wpcom/v2/',
+} );
+
+const mapSiteIdToSite = ( siteId: number | null ) => ( {
+	isCreated: !! siteId,
+	id: siteId ?? 0,
+} );
+
+const getGlobalStylesInfoForSite = ( site: Site ) => {
+	if ( ! site.isCreated ) {
+		return NEW_SITE_GLOBAL_STYLES_INFO;
+	}
+
+	const globalStylesInfoEndpoint = getGSInfoEndpointForSite( site.id );
+	return wpcom.req.get( globalStylesInfoEndpoint );
```

Comment:
The `useQuery` library handles success and error cases. On success, since we don't need any special mapping, we can just get away without any specific logic. If an error occurs, the `useQuery` retries the request, it does it with some sort of `exponential backoff`, but linear backoff since the timing function is f(t) = 2*t and the domain of t is [0,4]. 

So thanks to `useQuery` I can get away without needing to do basically anything.